src/screens/game_map.py

```diff
@@ -0,0 +1,681 @@
+        will be used when warping to farm_new should be named
+        "to farm_new".
+
+        :param pos: Position of the warp
+        :param obj: TiledObject to create the warp from
+        """
+        name = obj.name
+        if name == "spawnpoint":
+            if self.player_spawnpoint:
+                print(f"WARNING: Multiple spawnpoints found "
+                      f"({self.player_spawnpoint}, {pos})")
+            self.player_spawnpoint = pos
+        else:
+            name = name.split(" ")
+            if len(name) == 2:
+                warp_type = name[0]
+                warp_map = name[1]
+                if warp_type == "from":
+                    self.player_entry_warps[warp_map] = pos
+                elif warp_type == "to":
+                    warp_hitbox = pygame.Surface(
+                        (obj.width * SCALE_FACTOR,
+                         obj.height * SCALE_FACTOR)
+                    )
+                    Sprite(
+                        (obj.x * SCALE_FACTOR,
+                         obj.y * SCALE_FACTOR),
+                        warp_hitbox,
+                        name=warp_map
+                    ).add(self.player_exit_warps)
+                else:
+                    print(
+                        f"WARNING: Invalid player warp \"{name}\""
+                    )
+            else:
+                print(f"WARNING: Invalid player warp \"{name}\"")
+
+    def _setup_npc(
+            self,
+            pos: tuple[int, int]
+    ):
+        """
+        Creates a new NPC sprite at the given position
+        """
+        return NPC(
+            pos=pos,
+            assets=ENTITY_ASSETS.RABBIT,
+            groups=(self.all_sprites, self.collision_sprites),
+            collision_sprites=self.collision_sprites,
+            apply_tool=self.apply_tool,
+            soil_layer=self.soil_layer,
+            emote_manager=self.npc_emote_manager,
+        )
+
+    def _setup_animal(
+            self,
+            pos: tuple[int, int],
+            obj: TiledObject
+    ):
+        """
+        Creates a new Animal sprite at the given position.
+        The animal type is determined by the object name, objects named
+        "Chicken" will create Chickens, objects that are named "Cow" will
+        create Cows.
+        """
+        if obj.name == "Chicken":
+            return Chicken(
+                pos=pos,
+                assets=ENTITY_ASSETS.CHICKEN,
+                groups=(self.all_sprites, self.collision_sprites),
+                collision_sprites=self.collision_sprites,
+            )
+        elif obj.name == "Cow":
+            return Cow(
+                pos=pos,
+                assets=ENTITY_ASSETS.COW,
+                groups=(self.all_sprites, self.collision_sprites),
+                collision_sprites=self.collision_sprites,
+            )
+        else:
+            print(f"Malformed animal object name \"{obj.name}\" in tilemap")
```

Comment:
Once again, a warning would be better, so that actually gets shown in the error stream instead of the regular output one (where you'd typically show normal output)